File: quicksort/src/main/java/local/example/outcome/OutcomeResource.java
package local.example.outcome;

import local.example.outcome.algorithm.Quicksort;
import local.example.outcome.model.Item;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.*;

@Path("/outcome")
public class OutcomeResource {

    private final Set<List<Item>> itemsSet = Collections.newSetFromMap(
            Collections.synchronizedMap(
                    new LinkedHashMap<>()
            )
    );

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String feedback() {
        return "-- outcome feedback --";
    }

    @GET
    @Path("/items")
    @Produces(MediaType.APPLICATION_JSON)
    public Response read() {
        if (itemsSet.isEmpty()) {
            return Response.ok().build();
        }
        return Response.ok(itemsSet).build();
    }

    @POST
    @Path("/sorting")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    public Set<List<Item>> sorting(List<Item> items) {
        Quicksort.quickSort(items);
        itemsSet.add(items);
        return itemsSet;
    }
}

File: rms/src/main/kotlin/local/example/outcome/OutcomeResource.kt
package local.example.outcome

import local.example.outcome.model.RootMeanSquare
import java.util.*
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/outcome")
class OutcomeResource {

    private val rootMeanSquares = Collections.newSetFromMap(
        Collections.synchronizedMap(
            LinkedHashMap<RootMeanSquare, Boolean>()
        )
    )

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun read(): Response? {
        return if (
            rootMeanSquares.isEmpty()
        ) Response.ok().build() else Response.ok(rootMeanSquares).build()
    }
}
